src/stream/stream/mod.rs

```diff
@@ -573,6 +575,37 @@ extension_trait! {
             Enumerate::new(self)
         }
 
+        #[doc = r#"
+            Creates a stream that is delayed before it starts yielding items.
+
+            # Examples
+
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use async_std::prelude::*;
+            use async_std::stream;
+            use std::time::Duration;
+
+            let mut s = stream::from_iter(vec![0u8, 1, 2]).delay(Duration::from_millis(200));
+
+            assert_eq!(s.next().await, Some(0));
+            assert_eq!(s.next().await, Some(1));
```

Comment:
Should we assert at least 200 millis have passed between the call to the first `next` and when it actually resolves?

Reply:
Yah! You're right. 
I added delay time test with this commit.
fb1fb6c903fca2b68871706b873f3f548148b4b4